Vector.c

```diff
@@ -200,15 +192,13 @@ void Vector_insertionSort(Vector* this) {
    assert(Vector_isConsistent(this));
 }
 
-static void Vector_checkArraySize(Vector* this) {
+static void Vector_resizeIfNecessary(Vector* this, int newSize) {
+   assert(newSize >= 0);
    assert(Vector_isConsistent(this));
```

Comment:
Minor optimization: Only check `Vector_isConsistent` once the resizing is about to happen.

Thus either leave this line and move the one below to the end of the `if` statement, or move this one to the start of the `if` block.

Reply:
good catch